File: 0001-two-sum/0001-two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int n = numsSize;
    int *a = nums;
    *returnSize = 2;
    int *res = (int *) malloc(2 * sizeof(int));
    for(int i=0;i<n-1;i++)
    for(int j=i+1;j<n;j++)
    if(a[i] + a[j] == target)
    {
        res[0] = i;
        res[1] = j;
        return res;
    }
    return res;
    
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] a, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<a.length;i++)
        {
            int m= target-a[i];
            if(map.containsKey(m))
            return new int [] {i,map.get(m)};
            map.put(a[i],i);
        }
        return null;
    }
}